src/motd.rs

```diff
@@ -0,0 +1,408 @@
+use std::fs::{create_dir_all, File};
+use std::io::{Seek, Write};
+use std::path::{Path, PathBuf};
+
+use anyhow::Result;
+use async_std::stream::StreamExt;
+use log::warn;
+use nix::errno::Errno;
+use nix::mount::MsFlags;
+use serde::{Deserialize, Serialize};
+
+use crate::broker::Topic;
+use crate::dut_power::OutputState;
+use crate::temperatures::Warning;
+use crate::usb_hub::OverloadedPort;
+use crate::WatchedTasksBuilder;
+
+#[cfg(feature = "demo_mode")]
+mod setup {
+    pub(super) const VAR_RUN_TACD: &str = "demo_files/var/run/tacd";
+    pub(super) const ETC: &str = "demo_files/etc";
+
+    /// umount stub for demo_mode that works without root permissions
+    ///
+    /// (by doing nothing).
+    pub(super) fn umount(_target: &std::path::Path) -> nix::Result<()> {
+        Err(nix::errno::Errno::EINVAL)
+    }
+
+    /// mount stub for demo_mode that works without root permissions
+    ///
+    /// (by doing nothing).
+    pub(super) fn mount(
+        _source: Option<&std::path::Path>,
+        _target: &std::path::Path,
+        _fstype: Option<&str>,
+        _flags: nix::mount::MsFlags,
+        _data: Option<&str>,
+    ) -> nix::Result<()> {
+        Ok(())
+    }
+}
+
+#[cfg(not(feature = "demo_mode"))]
+mod setup {
+    pub(super) use nix::mount::{mount, umount};
+    pub(super) const VAR_RUN_TACD: &str = "/var/run/tacd";
+    pub(super) const ETC: &str = "/etc";
+}
+
+use setup::*;
+
+#[derive(Clone, Serialize, Deserialize)]
+struct MotdContent {
+    dut_pwr_state: OutputState,
+    iobus_fault: bool,
+    rauc_should_reboot: bool,
+    rauc_update_urls: Vec<String>,
+    setup_mode_active: bool,
+    temperature_warning: bool,
+    usb_overload: Option<OverloadedPort>,
+}
+
+pub struct Motd {
+    path_etc_motd: PathBuf,
+}
+
+const COLOR_RED: &str = "\x1b[31m";
+const COLOR_GREEN: &str = "\x1b[32m";
+const COLOR_YELLOW: &str = "\x1b[33m";
+const COLOR_RESET: &str = "\x1b[0m";
+
+impl MotdContent {
+    fn write(&self, dst: &mut File) -> std::io::Result<()> {
+        writeln!(dst, "Welcome to you TAC!")?;
+        writeln!(dst)?;
+
+        if self.temperature_warning {
+            writeln!(
+                dst,
+                "- {}WARNING{}: Your TAC is overheating, please provide proper airflow and let",
+                COLOR_RED, COLOR_RESET
+            )?;
+            writeln!(dst, "  it cool down.")?;
+        }
+
+        if self.setup_mode_active {
+            writeln!(
+                dst,
+                "- {}GREAT!{} You have logged in successfully, now you should continue the setup",
```

Comment:
Someone may not know that their device is is in setup mode, when SSHing into it.
Maybe add a note in the lines of `Device in setup mode`?

Reply:
Makes sense. I've changed the text to this now:

```
- GREAT! You have logged in successfully!
  Now you should continue the setup process in the web interface
  to leave the setup mode.
```

What do you think? I've tried to keep the message positive (e.g. not start with "WARNING: Device in Setup mode") to make the first impression of the device shell not an error message.

Reply:
I like the idea to keep it positive. But it also makes clear, that the device is in a somewhat special state. (Just in case the user does not know the device is in setup mode when logging in.)